### PR TITLE
New: Add automation conditions and actions

### PR DESCRIPTION
**Pull request by ericrpowers**
#### Description
As we start to resurrect the automation work, we want to make things more modular. Instead of automation templates, we will not have automation conditions and automation actions. This will allow us to more easily define the individual components of an automation in the future.

### CODE REVIEW

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "conditions-and-actions",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "cf52791bfe6716d7c17d2cc6558e9d02320949c0",
    "pipeline": "master"
  },
  {
    "project": "conditions-and-actions",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "21154f465241c08917e9889e55bc6498ed0ba9c6",
    "pipeline": "master"
  },
  {
    "project": "conditions-and-actions",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "bb35372b65973d00701b99f887a5ce28ada1b34c",
    "pipeline": "master"
  },
  {
    "project": "conditions-and-actions",
    "component": "cookbook",
    "pr-id": "38",
    "commit-sha": "cca6d1682a0fa9d2b039a51b2fd9e153b136007c",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "bb35372b65973d00701b99f887a5ce28ada1b34c",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "apotests",
    "pr-id": "3544",
    "commit-sha": "8536c617c077106ed6db0f694e1b188aa791bc52",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "bb35372b65973d00701b99f887a5ce28ada1b34c",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "apotests",
    "pr-id": "3544",
    "commit-sha": "47e108ff7aae1686790dbee4453f90713f8dd6e1",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "21154f465241c08917e9889e55bc6498ed0ba9c6",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "bb35372b65973d00701b99f887a5ce28ada1b34c",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "2b34b6dc6ca461c1cb1e7f03bdb278f7438af900",
    "pipeline": "master"
  }
]
```

File: specs/automationaction.spec [hunk 1]
```diff
@@ -26,36 +38,32 @@ attributes:
...
-
-  - name: kind
-    description: Represents the kind of template.
-    type: enum
-    exposed: true
-    allowed_choices:
-    - Action
-    - Condition
-    default_value: Condition
+    stored: true
```
Comment (t00f) [inline specs/automationaction.spec:53]: I am wondering if we still need this `key`. If it is required, we probably need an index.
cc @primalmotion

File: specs/automationcondition.spec [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  - '@migratable'
+  - '@base'
+  - '@namespaced'
+  - '@described'
+  - '@identifiable-stored'
+  - '@named'
+  - '@metadatable'
+  - '@propagated'
+  - '@timeable'
+
```
Comment (primalmotion) [inline specs/automationcondition.spec:32]: needs unique index on the (namespace,key)

File: specs/automationaction.spec [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
   - '@described'
+  - '@identifiable-stored'
   - '@named'
-
-# Indexes
-indexes:
-- - :no-inherit
+  - '@metadatable'
+  - '@propagated'
+  - '@timeable'
```
Comment (primalmotion) [inline specs/automationaction.spec:31]: needs unique index on the (namespace,key)

**Review by primalmotion — CHANGES_REQUESTED** [review]

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "d84dec33d6412fd64c177e986d23272ec1723ac9",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "3fac91fe0119ba4d03b03d9ed492fef02290c00b",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "3fac91fe0119ba4d03b03d9ed492fef02290c00b",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "3fac91fe0119ba4d03b03d9ed492fef02290c00b",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "0d6207036c2ea9a90f7bcf094f883be813dcda7b",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "19415a8630ace44d2a9f8a795dddcb347ca9bacd",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "9ab764664c6a17bb1ed329cb225849f770bd14b3",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  }
]
```

**Review by primalmotion — APPROVED** [review]

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "underwater",
    "pr-id": "180",
    "commit-sha": "fa702beda2f2386d07196c5e1a9d6db80f811df8",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "740",
    "commit-sha": "0ff921ec880a01b350f74a3b9446a9c5a90192d0",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "1447",
    "commit-sha": "45e8c79473ac326424517c4a3c058e8947767135",
    "pipeline": "master"
  }
]
```

**Pull request merged**